velox/functions/prestosql/window/AggregateWindow.cpp

```diff
@@ -113,16 +115,46 @@ class AggregateWindowFunction : public exec::WindowFunction {
     auto rawFrameEnds = frameEnds->as<vector_size_t>();
 
     FrameMetadata frameMetadata =
-        fillArgVectors(rawFrameStarts, rawFrameEnds, numRows);
+        computeFrameMetadata(rawFrameStarts, rawFrameEnds, numRows);
+
     if (frameMetadata.fixedFirstRow && frameMetadata.nonDecreasingFrameEnd) {
+      // If the previous block also had the same firstRow as the current block
```

Comment:
It would be more efficient (and less code) to pass BoundType to the AggregateWindowFunction's constructor.

Reply:
We are planning to enhance AggregateWindow to use advance/retract style aggregation. I feel that we can revise the API to also pass boundType to help at that point.

Given how the current code is written I feel like this change is only a continuation of the existing optimization idea. Since benchmark queries are just using UNBOUNDED PRECEDING AND CURRENT ROW frames, this is an easy change to reach a point where only the minimal work for that frame type is being done.

Reply:
I've made some changes to simplify the conditions in the code now. This current version reads better I think.